charts/linkerd2/templates/controller.yaml

```diff
@@ -72,6 +72,10 @@ spec:
         {{.ControllerNamespaceLabel}}: {{.Namespace}}
         {{- include "partials.proxy.labels" .Proxy | nindent 8}}
     spec:
+      {{- if .NodeSelector }}
```

Comment:
It'd be nice to DRY this up by having it be a template function. Check out how `_affinity.tpl` is doing it. Should collapse all of this down to a single call.

Reply:
Done!